src/thinblock.cpp

```diff
@@ -1341,15 +1244,15 @@ bool CThinBlockData::CheckThinblockTimer(uint256 hash)
     LOCK(cs_mapThinBlockTimer);
     if (!mapThinBlockTimer.count(hash))
     {
-        mapThinBlockTimer[hash] = GetTimeMillis();
+        mapThinBlockTimer[hash] = getTime();
         LogPrint("thin", "Starting Preferential Thinblock timer\n");
     }
     else
     {
         // Check that we have not exceeded the 10 second limit.
         // If we have then we want to return false so that we can
         // proceed to download a regular block instead.
-        uint64_t elapsed = GetTimeMillis() - mapThinBlockTimer[hash];
+        uint64_t elapsed = getTime() - mapThinBlockTimer[hash];
         if (elapsed > 10000)
```

Comment:
Also removed  LOCK(cs_thinblockstats); from ::updateStats() as that was superfluous.